qcodes/tests/dataset/test_measurement_context_manager.py

```diff
@@ -1478,18 +1480,18 @@ def test_datasaver_array_parameters_array(channel_array_instrument, DAC, N,
         expected_sp_data = expected_sp_data.reshape(N, M)
         expected_output = expected_output.reshape(N, M)
 
-    assert_allclose(data['dummy_dac_ch1'], expected_dac_data)
-    assert_allclose(data['dummy_channel_inst_ChanA_this_setpoint'],
+    assert_allclose(loaded_data['dummy_dac_ch1'], expected_dac_data)
+    assert_allclose(loaded_data['dummy_channel_inst_ChanA_this_setpoint'],
                     expected_sp_data)
-    assert_allclose(data['dummy_channel_inst_ChanA_dummy_array_parameter'],
+    assert_allclose(loaded_data['dummy_channel_inst_ChanA_dummy_array_parameter'],
                     expected_output)
 
     if storage_type == 'array':
         # for now keep testing the old way of getting data (used by
         # plot_by_id). Hopefully this will eventually be deprecated
```

Comment:
```suggestion
        # also test get_data_by_id (used by plot_by_dataset)
```

Reply:
We should still deprecate this one  get_data_by_id eventually but sure we can remove it from here